### ISSUE TITLE
Unable to reindex after stuck

### ISSUE THREAD

**Issue by unional** [HvyIndustries/crane#193]
This may be lower priority.

When indexing was stuck, issue reindex or changing the exclude paths results in:
```sh
rejected promise not handled within 1 second

shell.ts:541 channel closed: Error: channel closed
    at ChildProcess.target.send (internal/child_process.js:540:16)
    at IPCMessageWriter.write (C:\Users\hwong\.vscode\extensions\HvyIndustries.crane-0.2.2\node_modules\vscode-jsonrpc\lib\messageWriter.js:34:22)
    at Object.sendNotification (C:\Users\hwong\.vscode\extensions\HvyIndustries.crane-0.2.2\node_modules\vscode-jsonrpc\lib\main.js:180:27)
    at Object.didChangeConfiguration (C:\Users\hwong\.vscode\extensions\HvyIndustries.crane-0.2.2\node_modules\vscode-languageclient\lib\main.js:52:71)
    at LanguageClient.onDidChangeConfiguration (C:\Users\hwong\.vscode\extensions\HvyIndustries.crane-0.2.2\node_modules\vscode-languageclient\lib\main.js:489:28)
    at LanguageClient.<anonymous> (C:\Users\hwong\.vscode\extensions\HvyIndustries.crane-0.2.2\node_modules\vscode-languageclient\lib\main.js:475:81)
    at e.invoke (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:31992)
    at e.fire (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:5:8945)
    at t.$acceptConfigurationChanged (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:10:22029)
    at t.e.handle (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:10:4203)
    at s (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:7:11178)
    at p (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:7:11855)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```

**Comment by NewFuture** [comment]
The same bug +1

**Comment by nevadascout** [comment]
There are some cases where it crashes so hard that it actually crashes the VS Code plugin server process for Crane. I'm still investigating this, but for now a temporary fix is to just restart Crane when this happens.

**Comment by nevadascout** [comment]
Fixed in v0.3.0
